src/plugins/Workers/SyncWorker/utils/TransactionOrderer/methods/insertAtPos.js

```diff
@@ -0,0 +1,17 @@
+/**
+ * Add a transaction at a specific position
+ * @param {Transaction} transaction
+ * @param {number|null} pos
+ */
+module.exports = function insertAtPos(transaction, pos) {
+  const { transactionIds, transactions } = this;
+  if (pos != null) {
+    // eslint-disable-next-line no-param-reassign
+    if (pos > transactions.length) pos = transactions.length;
+    transactions.splice(pos, 0, transaction);
+    transactionIds.splice(pos, 0, transaction.hash);
+  } else {
```

Comment:
`else` usually complicates readability. We could make it probably cleaner:

```js
if (pos === null || pos > transactions.length) {
    transactions.push(transaction);
    transactionIds.push(transaction.hash);

    return this;
}

transactions.splice(pos, 0, transaction);
transactionIds.splice(pos, 0, transaction.hash);

return this;
```

Reply:
Fixed for the better suggestion!